### DISCUSSION TITLE
Just a quick query

### DISCUSSION THREAD

**Discussion by chuck808** [chhorisberger/M5StackMenuSystem#5] [General]
Hi,

First of all, thank you for making this library available, it's awesome.

My query, is there any way to get it to work using the M5Core2.h library instead of the M5Unified.h? I only ask because I would like to use it in an application I'm working on, but everything I have done so far uses the M5Core2 library. If not, no worries, I still think this libray is ecellent.

Regards

**Comment by lvlhead** [comment]
> Hi,
> 
> First of all, thank you for making this library available, it's awesome.
> 
> My query, is there any way to get it to work using the M5Core2.h library instead of the M5Unified.h? I only ask because I would like to use it in an application I'm working on, but everything I have done so far uses the M5Core2 library. If not, no worries, I still think this libray is ecellent.
> 
> Regards

Is there a feature you use in M5Core2 library that is not available in M5Unified?  I took a cursory glance at your GH project and noticed it looks fairly easy to port to Unified.

In case it's not possible, you could probably edit this library in your Arduino folder to use the M5Core2 headers.  I helped port it to Unified, you can look at the changes made in my fork (or the commits between library version 1.1.3 and 1.2.1) and use it as a basis for yours.

Best of luck!

**Reply by chhorisberger** [reply]
Thanks for input @lvlhead - I totally agree with you.

@chuck808 Unless there's some unexpected complications, upgrading your project from M5Core2 to U5Unified should probably just be a matter of running a "replace all" in your project.

replacing 
`#include <M5Core2.h>`
with
`#include <M5Unified.h>`

If there are really some features you are using in M5Core2 which have no replacement in M5Unified you could of course download and edit this library and do the opposite, i.e.  replacing

`#include <M5Unified.h>`
with
`#include <M5Core2.h>`

But if possible, I would recommend against it, since you would then be cut out from future updates of this library.
Also you'll exclude out all the non-M5Core2 owners to use your own library ;)

**Reply by chuck808** [reply]
Thank you for the reply, and apologies for the delayed response. 

I had not intended to use it for the project I have on GH. I am working on something else that has taken quite a while to get round to finishing, so I can't even remeber the reason for not using M5Unified.

I seem to remeber I as adding audio to the project and M5Unified was a better option for that. The when I tried to something else (the bit I can't remember :) )I had to change over to the M5Core2 library, which was a pain. So I was trying to avoid that messing around again.

I suppose I could make a copy of the project and just try it with M5Unified, which would obviously solve the isuue or at least answer my query.

Regards
